### PR TITLE
#167899197 Users Should Be Able To Read Articles

### PR DESCRIPTION
**Pull request by JuwonAbiola**
#### What does this PR do?
- Enables a user to view an article
#### Description of Task proposed in this pull request?
- Create viewArticle method
- Update Swagger Docs
- Write test to test viewArticle method

#### How should this be manually tested (Quality Assurance)?
- Explain how to manually test this feature. For example if changes were made in
the UI or in the API, explain where and if any specific access is needed.

#### What are the relevant pivotal tracker stories?
- [167899197](https://www.pivotaltracker.com/story/show/167899197)

#### Any background context you want to add (Operations Impact)?
- The endpoint for view article is not protected 
- Run `npm run db:rollback` the article seeder file was updated..

#### What I have learned working on this feature: 
- I learned how to stub 500 error

#### Screenshots: 
N/A

### CODE REVIEW

File: server/controllers/Articles.js [hunk 1]
```diff
@@ -335,6 +335,57 @@ const createLikeOrDislike = async (userAction, userId, article) => {
...
+   * @returns {JSON} JSON object with details of new article
+   */
+  static async viewArticle(req, res) {
+    try {
+      const {
+        params: { slug }
+      } = req;
+      const article = await Article.findBySlug(slug);
+      if (!article || article.isArchived) {
+        return serverResponse(res, 404, { error: 'article not found' });
```
Comment (chialuka) [inline server/controllers/Articles.js:540]: Shouldn't a logged in user be able to view his own archived articles?

File: test/articles/viewArticle.test.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+config();
+
+chai.use(chaiHttp);
+
+const { BASE_URL } = process.env;
+
+describe('GET Article', () => {
+  context('when a user views a published article', () => {
+    it('it returns the user profile', async () => {
```
Comment (chialuka) [inline test/articles/viewArticle.test.js]: There is an extra "it" in your it block

File: test/articles/viewArticle.test.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        'publishedAt',
+        'createdAt',
+        'updatedAt',
+        'authorId'
+      );
+    });
+  });
+
+  context('when a user tries to view an article that does not exist', () => {
+    it('it returns a not found error', async () => {
```
Comment (chialuka) [inline test/articles/viewArticle.test.js]: ditto

File: server/controllers/Articles.js [hunk 4]
```diff
@@ -335,6 +335,57 @@ const createLikeOrDislike = async (userAction, userId, article) => {
...
+      }
+      const {
```
Comment (chialuka) [inline server/controllers/Articles.js]: Is there a reason for de-structuring the article object?

**Review by dinobi — DISMISSED** [review]

**Comment by ayodejiAA** [comment]
@jhayX, the return does not contain 'following'

**Review by dinobi — APPROVED** [review]

**Pull request merged**